### PR TITLE
Updates to the rust simple example.

### LINKED ISSUE #703: Proposal: Split up the api group stable.agones.dev
This has been rolling around in my head for a while, so wanted to throw it out there.

Basically watching the changes coming to allocation (#682), and how that will _have_ to be in a different group, and then the multicluster work (#698) feels like it should be separated in someway from everything else - I'm wondering if having `stable.agones.dev` as the core group was a less than ideal decision.

If we made this change, I a sacrificial draft of grouping could be:

core.agones.dev:
 - GameServers
 - Fleets
 - GameServerSets

scaling.agones.dev
 - FleetAutoscaling

allocation.agones.dev
 - GameServerAllocation

multicluster.agones.dev
 - AllocationPolicy (or whatever it ends up being called)

(FleetAllocation would probably have to stay in core, until it goes away, just because you can't mix CRDs and APIServices)

Looking at Kubernetes - they also do something similar.

This is obviously a breaking change - so wanted to run it past everyone, see how people felt.

### PR DESCRIPTION
**Pull request by roberthbailey**
Fixes #703.

I re-ordered things into what I thought was a more useful ordering (less work -- cloud -- to more work -- local dev environment) and made it so that we could push an "official" image to the `agones-images` project so that you can run the example without needing to compile any code. 

One thing that is weird is that while the running locally with docker instructions work for me, when I run the sample in an actual GKE cluster the game server never moves to Ready -- even though the sidecar is getting health pings the game server never transitions out of the unhealthy state. Maybe @aLekSer has an idea of why this might be?

### CODE REVIEW

**Comment by aLekSer** [comment]
Re GKE cluster I cannot find any images under
https://console.cloud.google.com/gcr/images/agones-images/GLOBAL/rust-simple-server?gcrImageListsize=30

**Comment by roberthbailey** [comment]
@aLekSer - there aren't any images there (yet). I've been testing with images in my own project so that I only push images to the official project once the code has been reviewed. 

From https://github.com/googleforgames/agones/blob/c1dd21fb90fc7b9c8ee9515789c1e2dd51e00830/examples/rust-simple/README.md:

If you want to build your own image

```bash
$ REPOSITORY=<your-repository> # e.g. gcr.io/agones-images
$ make build-image REPOSITORY=${REPOSITORY}
$ docker push ${REPOSITORY}/rust-simple-server:0.3
```

To run your own image, make a copy of
[examples/rust-simple/gameserver.yaml](https://raw.githubusercontent.com/googleforgames/agones/master/examples/rust-simple/gameserver.yaml),
modify the image name to the image you pushed, and run `kubectl create -f gameserver.yaml`.

**Comment by aLekSer** [comment]
@roberthbailey thanks for clarification, just want to let you note that the original gcr.io/agones-images/rust-simple-server:0.3 should be pushed at some point for others to follow `README.md`.

**Comment by roberthbailey** [comment]
Definitely. I just want to make sure that it works first (right now it doesn't -- see https://github.com/googleforgames/agones/issues/938).

**Comment by aLekSer** [comment]
Yes, I am new to Rust language that's why I can not provide a fix for #938 . Seems that it is needed to add some loop with sleep at the beginning on SDK connect stage.

File: examples/rust-simple/README.md [hunk 1]
```diff
@@ -8,62 +8,70 @@ It will
...
-```
-cd ../../build; make run-sdk-conformance-local TIMEOUT=120 TESTS=ready,watch,health,gameserver
-```
-This will start an SDK-server in a docker, which will be running for 120 seconds.
 
-Run the Rust Simple Gameserver binary:
+To run the example server:
 ```
-make run
+kubectl create -f https://raw.githubusercontent.com/googleforgames/agones/master/examples/rust-simple/gameserver.yaml
```
Comment (markmandel) [inline examples/rust-simple/README.md]: Same as the C++ one, should we make this 
```suggestion
kubectl create -f https://raw.githubusercontent.com/googleforgames/agones/<release-branch>/examples/rust-simple/gameserver.yaml
```

With a note saying "Where <release-branch> should match the Agones version you have deployed, e.g `release-0.12`"

Sound good?

**Comment by roberthbailey** [comment]
> I assume we aren't going to merge this until we fix #938 ?

Yes. Unless it looks like it won't be fixed before the 0.12.0 release in which case I'll cut out the part about running in a cluster and replace it with a warning that the rust SDK is still experimental.

**Comment by markmandel** [comment]
Just to confirm - this is building now, but just isn't blocking? If that;s the case, I'll approve so we at least have an example that works.

**Comment by roberthbailey** [comment]
It builds and runs following the docs. But because the container doesn't block, when running the example on GKE the pod restarts and that means that the gameserver never becomes healthy.

**Comment by markmandel** [comment]
Got it - thanks! I'll take a look, see if I can work out a quick solution as well. I've used rust before in a previous life, so hopefully it will come back to me 😄 

Locally. I've got some changes to the conformance tests that will also check for the blocking functionality going forward.

**Comment by roberthbailey** [comment]
And running locally works just fine.

**Comment by markmandel** [comment]
Once #953 is merged, this should be good to go.

**Comment by roberthbailey** [comment]
I'm going to re-work this to create a new tutorial like the cpp/nodejs ones. Changes should be ready shortly.

File: site/content/en/docs/Tutorials/simple-gameserver-rust.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
+---
```
Comment (markmandel) [inline site/content/en/docs/Tutorials/simple-gameserver-rust.md]: I love these :blue_heart:

**Pull request merged**